src/krb5/_localauth.pyx

```diff
@@ -0,0 +1,47 @@
+# Copyright: (c) 2022 Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+from libc.stdlib cimport free, malloc
+
+from krb5._exceptions import Krb5Error
+
+from krb5._context cimport Context
+from krb5._krb5_types cimport *
+from krb5._principal cimport Principal
+
+
+cdef extern from "python_krb5.h":
+
+    krb5_error_code krb5_aname_to_localname(
+        krb5_context context,
+        krb5_const_principal aname,
+        int lnsize,
+        char *lname_out
+    ) nogil
+
+    krb5_error_code KRB5_LNAME_NOTRANS
+
+def aname_to_localname(
+    context: Context,
+    principal: Principal
+) -> str:
+
+    cdef krb5_error_code err = 0
+    limit = 256
+
+    cdef char *localname = <char *>malloc(limit + 1)
+    if not localname:
+        raise MemoryError()
+
+    err = krb5_aname_to_localname(context.raw, principal.raw, limit, localname)
+    # The definition of KRB5_LNAME_NOTRANS in the Heimdal header file
+    # included in the pykrb5 source appears to be out of date; this is
+    # the value that actually gets returned in the test -- so I'm just
+    # adding it here for now.
+    if err in [KRB5_LNAME_NOTRANS, -1765328227]:
```

Comment:
Looks like `-1765328227` is `KRB5_NO_LOCALNAME`, it's not surprising that Heimdal is returning a different value than MIT. You can even see it in https://github.com/heimdal/heimdal/blob/fa92fe37e7ed1cfbe546fd617d712f47c55e35b9/lib/krb5/aname_to_localname.c#L330 that it is returning `KRB5_NO_LOCALNAME`. It's been in MIT krb5 and Heimdal for over 10 years so it's probably safe to for years so it's probably safe to do `if err in [KRB5_LNAME_NOTRANS, KRB5_NO_LOCALNAME]:` with a brief comment as to why we check both.

Just as an FYI the Heimdal headers in the repo are only used for macOS builds as they are not available on the host. If compiling against Heimdal on an actual Linux host it will use the headers from the source you are linking against.